allennlp/common/params.py

```diff
@@ -490,6 +493,8 @@ def from_file(
 
         file_dict = json.loads(evaluate_file(params_file, ext_vars=ext_vars))
 
+        if isinstance(params_overrides, dict):
+            params_overrides = json.dumps(params_overrides)
         overrides_dict = parse_overrides(params_overrides)
```

Comment:
How about just:

```suggestion
        if isinstance(params_overrides, dict):
            overrides_dict = params_overrides
        else:
            overrides_dict = parse_overrides(params_overrides)
```

Reply:
Yeah, I actually tried this first. But there is some logic to `parse_overrides` that converts the value.

For example, let's say we have the following as the input:
```python
input_ = {
    'train_data_path': 'FOO',
    'model': {'type': 'BAR'},
    'model.text_field_embedder.tokens.type': 'BAZ',
    'data_loader.batch_sampler.sorting_keys.0': 'question'
}
```

This happens if we pass it into `parse_overrides` after `dump`ing:
```python
parse_overrides(json.dumps(input_))
# prints {'data_loader': {'batch_sampler': {'sorting_keys': {'0': 'question'}}},
#         'model': {'type': 'BAR', 'text_field_embedder': {'tokens': {'type': 'BAZ'}}},
#         'train_data_path': 'FOO'}
```

I want `parse_overrides` to do the same kind of transformation of the `dict` as it would do if it were represented as a JSON string. Does this make sense?

Reply:
Oh right, that's a good point.